datastructs/ring.h

```diff
@@ -9,20 +19,87 @@
 
 struct ring{
         /* Fill this in */
+        int size;
+        int pop;
+	int push;
+        ptr_t *items;
+}
+
+int isEmpty(struct ring *rb)
+{
+	return ( rb->pop == rb->push );
+}
+
+int isFull(struct ring *rb)
+{
+	return   ( rb->pop + 1 ) % rb->size == rb->push;
 }
 
+
 /* Create a ring buffer with the specified size. Return the ring or NULL
    if there is not enough memory to create. */
-struct ring* ring_create(int size);
+struct ring* ring_create(int size)
+{
+	
+	struct ring *rb = (struct ring*) calloc (1 ,sizeof(struct ring ));
+
+	if (rb == NULL ) 
+	{
+	return NULL;
```

Comment:
check and fix your indentation--the style guide requires you to use "soft tabs" made up of 8 spaces per indentation level. You should learn how to adjust your editor to enforce this.  I think right now you have a mix of spaces and real tab symbols.